### PR TITLE
Plugin support for EvaluatableExpressionFilter

### LINKED ISSUE #13454: Add plugin support to EvaluatableExpressionFilter
2.2 is introducing plugins, which can extend type mappings and method/member translations from outside EF Core. I'm currently porting Npgsql's own plugin model (introduced in 2.1) to the new EF Core one (https://github.com/npgsql/Npgsql.EntityFrameworkCore.PostgreSQL/issues/658), and one missing extension point is for `EvaluatableExpressionFilter`: our NodaTime plugin flags `SystemClock.Instance.GetCurrentInstance()` as non-evaluatable, to make sure it's evaluated at the server (much like `DateTime.Now`).

It seems important for this to make it into 2.2, otherwise NodaTime specifically won't be portable to the new plugin model (and this may also affect other plugins). I'll probably be able to submit a PR in about 2-3 weeks, is that early enough?

### PR DESCRIPTION
**Pull request by roji**
Closes #13454

### CODE REVIEW

File: src/EFCore/Query/IEvaluatableExpressionFilterPlugin.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    /// </summary>
+    public interface IEvaluatableExpressionFilterPlugin
+    {
+        /// <summary>
+        ///     Checks whether the given expression can be evaluated.
+        /// </summary>
+        /// <param name="expression"> The expression. </param>
+        /// <param name="model"> The model. </param>
+        /// <returns> True if the expression can be evaluated; false otherwise. </returns>
+        bool IsEvaluatableExpression([NotNull] Expression expression, [NotNull] IModel model);
```
Comment (smitpatel) [inline src/EFCore/Query/IEvaluatableExpressionFilterPlugin.cs]: no model.

Comment (roji) [reply src/EFCore/Query/IEvaluatableExpressionFilterPlugin.cs]: Make sense, done.

**Comment by smitpatel** [comment]
You did not squash again. 😢

**Comment by roji** [comment]
A build break *and* a forgotten squash...

image

**Pull request merged**